src/shell/mod.rs

```diff
@@ -1723,12 +1723,12 @@ impl Shell {
                 let geo = o.geometry();
                 match direction {
                     Direction::Left | Direction::Right => {
-                        !(geo.loc.y + geo.size.h < current_output_geo.loc.y
-                            || geo.loc.y > current_output_geo.loc.y + current_output_geo.size.h)
+                        geo.loc.y < current_output_geo.loc.y + current_output_geo.size.h
```

Comment:
Flipping the logic around helped avoid the issue where SwitchOutput to left resulted in output switch where it shouldn't as my external monitor is above my laptop regarding display.